lib/portlayer/metrics/metrics.go

```diff
@@ -34,18 +38,19 @@ var (
 // super manages the lifecycle and access to the
 // available metrics collectors
 type super struct {
-	vms *collectorVM
+	vms *performance.VMCollector
 }
 
-type Collector interface {
-	// Subscribe to a stream of this collectors metrics
-	Subscribe(interface{}) (chan interface{}, error)
-	// Unsubscribe to a stream of this collectors metrics
-	Unsubscribe(interface{}, chan interface{})
-	// SubscriberCount returns the number of subscribers for the collector
-	SubscriberCount() int
-	// Sample will return metrics without a subscription
-	Sample(interface{}) (chan interface{}, error)
+type UnsupportedTypeError struct {
+	subscriber interface{}
+}
+
+func (ute UnsupportedTypeError) Error() string {
+	t := reflect.TypeOf(ute.subscriber)
+	if t.Kind() == reflect.Ptr {
+		t = t.Elem()
+	}
+	return fmt.Sprintf("type %s is not supported by metrics", t.Name())
```

Comment:
I've changed it...to just the type...so `fmt.Sprintf("type %s is not supported by metrics", t)`